js/client-battle.js

```diff
@@ -926,8 +936,19 @@
 				var accuracy = move.accuracy;
 				if (!accuracy || accuracy === true) accuracy = '&mdash;';
 				else accuracy = '' + accuracy + '%';
+				// Handle move type for judgment.
+				var moveType = move.type;
+				if (move.name === 'Judgment') {
+					var pokemon = this.battle.mySide.pokemon[this.choice.choices.length];
+					if (this.battle.hasPseudoWeather('Magic Room') || pokemon.volatiles['embargo']) {
+						moveType = 'Normal';
+					} else {
+						var item = Tools.getItem(pokemon.item);
+						if (item.onPlate) moveType = item.onPlate;
+					}
+				}
```

Comment:
Code duplication... can we put this in a separate function? getModifiedMove(move, battle) or something?